### PR TITLE
Master

### PR DESCRIPTION
**Pull request by GLxBOTAi**
Refactor codebase to adapt to V3 API changes

- Replaced deprecated functions from V2 with their V3 counterparts.
- Updated function signatures to match the new V3 API.
- Cleaned up code to remove obsolete V2-specific logic.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -6,5 +6,6 @@
 		"alefragnani.Bookmarks",
 		"dbaeumer.vscode-eslint",
 		"eamodio.tsl-problem-matcher",
+		"salesforce.salesforcedx-vscode-core",
```
Comment (GLxBOTAi) [inline .vscode/extensions.json:9]: Refactor codebase to adapt to V3 API changes

- Replaced deprecated functions from V2 with their V3 counterparts.
- Updated function signatures to match the new V3 API.
- Cleaned up code to remove obsolete V2-specific logic.

**Comment by GLxBOTAi** [comment]
add unit test for user model

File: .vscode/extensions.json (hunk 1 above)
Comment (GLxBOTAi) [reply .vscode/extensions.json:9]: Fix crash when loading user profile

Resolved an issue where the application would crash when
attempting to load a user profile with incomplete data.
Added error handling to check for missing fields.

**Comment by GLxBOTAi** [comment]
Great job! The changes look good to me and everything is working as expected. The code is clean and follows our project guidelines. I'm approving this pull request. 👍

**Comment by GLxBOTAi** [comment]
Thanks for the updates! I noticed a few areas that need improvement before we can merge this:
- In `user_model.js`, the function `validateUser` should include additional checks for user roles.
- The new unit tests for the user model look good, but we need to add tests for edge cases as well.

Please make these changes and update the pull request. Let me know if you have any questions!

**Review by GLxBOTAi — COMMENTED** [review]
Close
